Framework/Interaction/Line.php

```diff
@@ -649,4 +651,29 @@ protected function isProductCalculated(\Magento\Sales\Api\Data\OrderItemInterfac
             return true;
         }
     }
+
+    /**
+     * @param $product
+     * @param integer $storeId
+     * @return array
+     */
+    protected function buildItemData($product, $storeId)
+    {
+        $data = [];
+        if ($product) {
+            $data['itemCode'] = $this->taxClassHelper->getItemCodeOverride($product);
```

Comment:
I would have created the `$data` array like this. However I'd say it's an optional refactor.

```
$data [
    'itemCode' => $this->taxClassHelper->getItemCodeOverride($product),
    'taxCode' => $this->taxClassHelper->getAvataxTaxCodeForProduct($product, $storeId),
    'productRef1' => $this->taxClassHelper->getRef1ForProduct($product),
    'productRef2' => $this->taxClassHelper->getRef2ForProduct($product)
]
```